File: 101. Symmetric Tree/Solution.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */


bool isSymmetricHelper(struct TreeNode* left,struct TreeNode* right)
{
  if(left==NULL && right==NULL )return true;
  if(left==NULL || right==NULL )return false;
  if(left->val!=right->val)return false;
  return isSymmetricHelper(left->right,right->left) && isSymmetricHelper(left->left,right->right);


}


bool isSymmetric(struct TreeNode* root){
if(root==NULL)return true;
return isSymmetricHelper(root->left,root->right);
}

File: 101. Symmetric Tree/Solution.js
// 101. Symmetric Tree
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */
function isSymmetric(root) {
    if (root === null)
        return true;
    return isSymmetricHelper(root.left, root.right);
}
;
function isSymmetricHelper(left, right) {
    if (left === null && right === null)
        return true;
    if (left === null || right === null)
        return false;
    if (left.val !== right.val)
        return false;
    return isSymmetricHelper(left.right, right.left) && isSymmetricHelper(left.left, right.right);
}
